### PR TITLE
Refactor code related to setting of default max heap size

### PR DESCRIPTION
**Pull request by ashu-mehra**
Changes in GC:
Added a virtual method MM_GCExtensionsBase::computeDefaultMaxHeap()
that sets MM_GCExtensionsBase::memoryMax. It uses new portlibrary API
sysinfo_get_addressable_physical_memory() to compute the heap size.

Changes in portlibrary APIs:
1. Updated sysinfo_cgroup_get_memlimit() to return negative error code
if the cgroup memory limit is not specified.
2. Added sysinfo_get_addressable_physical_memory() that returns physical
memory based on limitations on address space.

Signed-off-by: Ashutosh Mehra <asmehra1@in.ibm.com>

### CODE REVIEW

**Comment by ashu-mehra** [comment]
fyi - @dmitripivkine @charliegracie @DanHeidinga @dinogun

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by ashu-mehra** [comment]
I have updated commit to fix omrporttest failures.
@charliegracie can you please trigger the builds again.

**Comment by charliegracie** [comment]
@genie-omr build all zos

**Comment by ashu-mehra** [comment]
@charliegracie can this be merged now?

File: port/win32/omrsysinfo.c [hunk 1]
```diff
@@ -773,6 +773,12 @@ omrsysinfo_get_memory_info(struct OMRPortLibrary *portLibrary, struct J9MemoryIn
 	return 0;
 }
 
+uint64_t
+omrsysinfo_get_addressable_physical_memory(struct OMRPortLibrary *portLibrary)
+{
+	return portLibrary->sysinfo_get_physical_memory(portLibrary);
```
Comment (charliegracie) [inline port/win32/omrsysinfo.c]: Are we not missing this check `if (OMRPORT_LIMIT_LIMITED == portLibrary->sysinfo_get_limit(portLibrary, OMRPORT_RESOURCE_ADDRESS_SPACE, &memoryLimit))`?

Comment (ashu-mehra) [reply port/win32/omrsysinfo.c]: I intentionally omitted the check as `sysinfo_get_limit()` on win32 always returns `OMRPORT_LIMIT_UNKNOWN_VALUE`.
Do you want me to add that check?

**Pull request merged**